File: 3'rd Semester/assig/inCPP/q8.cpp
#include "Stack.cpp"
#include <iostream>

using namespace std;

bool findpl(string plStr, int i) {
    if (i > plStr.size() / 2) {
        return true;
    }
    return plStr[i] == plStr[plStr.size() - i - 1] && findpl(plStr, i + 1);
}

int main() {
    char plStr[] = "noon";
    if(findpl(plStr, 0)) {
        cout << "Yes";
    } else {
        cout << "No";
    }
}

File: 3'rd Semester/assig/q8.java
public class q8 {

    public static boolean findpl(String pl) {
        int len = pl.length();

        if (len < 2) {
            return true;
        }

        if (pl.charAt(0) != pl.charAt(len - 1)) {
            return false;
        }

        return findpl(pl.substring(1, len - 1));
    }

    public static void main(String[] args) {
        String pl = "noon";
        System.out.println(findpl(pl));
    }
}
